docs/user/sbt.rst

```diff
@@ -183,6 +183,25 @@ Use the following approach in `sbt` to set the target triple:
 
     nativeConfig ~= { _.withTargetTriple("x86_64-apple-macosx10.14.0") }
 
+Build target
+------------
+
+Setting build target allows you to specify to what type of object your project should be linked to.
+As an example, to link it as dynamic library use the following command:
+
+.. code-block:: scala
+
+    nativeConfig ~= { _.withBuildTarget(BuildTarget.libraryDynamic) }
+
+1. **application** (default)
+
+   Results in creating ready to use executable program.
+
+2. **libraryDynamic**
+
+   Results in dynamic library being built based on entry point methods annotated with `@export`,
+   for details see :ref:`interop`.
```

Comment:
how about producing a header file too.

Reply:
We can do this in either following PR or in seperate project outside core repo. I opt for the latter one. All informations needed to do this are contained in NIR format so we can e.q.: make it a part of `scala-native-bindgen`.